File: exo3.2.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>


int main (int argc, char** argv) {

    if (argc != 2) {
        fprintf (stderr, "Veuillez entrez un seul argument.\n");
        exit (0);
    }

    char* cp = "/usr/bin/cp";
    char* rep = argv[1];
    char* arg[] = {cp, "-r", ".", rep, NULL};

    execv (cp, arg);

    return 0;
}

File: exo3.2.py
#!/usr/bin/env python3
import sys
import os
import subprocess

# Programme qui exécute cp -r . <répertoire>

def main():
    if len(sys.argv) != 2:
        print("Veuillez entrer un seul argument.", file=sys.stderr)
        sys.exit(0)
    
    rep = sys.argv[1]
    
    # Utilisation de subprocess.run pour remplacer execv
    # Cela équivaut à exécuter: cp -r . rep
    result = subprocess.run(["/usr/bin/cp", "-r", ".", rep])
    
    # Le processus Python continue après l'exécution (contrairement à execv)
    # Pour imiter exactement execv, on pourrait utiliser os.execv mais cela
    # remplacerait le processus Python

if __name__ == "__main__":
    main()

